lib/common/asar.js

```diff
@@ -217,6 +224,28 @@
       arguments[arg] = newPath
       return old.apply(this, arguments)
     }
+    if (old[util.promisify.custom]) {
+      module[name][util.promisify.custom] = function () {
+        const p = arguments[arg]
+        const [isAsar, asarPath, filePath] = splitPath(p)
+        if (!isAsar) {
+          return old[util.promisify.custom].apply(this, arguments)
+        }
+
+        const archive = getOrCreateArchive(asarPath)
+        if (!archive) {
+          return new Promise(() => invalidArchiveError(asarPath))
```

Comment:
This should be `return Promise.reject(invalidArchiveError(asarPath))`

Reply:
@MarshallOfSound that won't work, `invalidArchiveError` throws the error

Reply:
Oh I see, this will result in a rejected promise right? 😄

Reply:
yes